src/main/java/uk/gov/hmcts/reform/slc/services/servicebus/MessageProcessor.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.reform.slc.services.servicebus;
+
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.slc.services.SendLetterJob;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@Component
+public class MessageProcessor {
+    private static final Logger logger = LoggerFactory.getLogger(SendLetterJob.class);
+
+    private final Supplier<IMessageReceiver> receiverProvider;
+
+    public MessageProcessor(Supplier<IMessageReceiver> receiverProvider) {
+        this.receiverProvider = receiverProvider;
+    }
+
+    /**
+     * Reads message from a queue and passes is as an argument to action.
+     */
+    public void handle(Function<IMessage, MessageHandlingResult> action) {
+
+        IMessageReceiver messageReceiver;
+
+        try {
+            messageReceiver = receiverProvider.get();
+        } catch (ConnectionException e) {
+            logger.error("Unable to connect to Service Bus", e);
```

Comment:
It's kind of wrapping same error report as provider creates one. could follow the pattern and catch same exceptions here and get rid off _wrapper_ exception

Reply:
unfortunately `InterruptedException` and `ServiceBusException` are checked exceptions, so I can't